var/spack/repos/builtin/packages/hwloc/package.py

```diff
@@ -60,6 +60,7 @@ class Hwloc(AutotoolsPackage):
     depends_on('libpciaccess', when='+pci')
     depends_on('libxml2', when='+libxml2')
     depends_on('pkg-config', type='build')
+    depends_on('libpciaccess', when=(sys.platform != 'darwin'))
```

Comment:
I was confused. The way things were handled in `hwloc` was correct after all:
- There is a variant `+pci` that defaults to `platform!=darwin`
- There is a dependency on `libpciaccess` depending on the `+pci` variant

This is correct because enabling PCI in hwloc unconditionally requires `libpciaccess`, so the conditional on the platform should be for the `pci` variant.

I have corrected this pull request.